mestopy/mestopy.py

```diff
@@ -1 +1,242 @@
+            new_device = DeviceObj(device_data,
+                                   sens,
+                                   device_name)
+            self.devices.append(new_device)
+        else:
+            # check if n_bins of all devices is the same
+            if not self.devices[0].data.n_bins == device_data.n_bins:
+                raise ValueError("ref_signal has wrong n_bins")
+            # check if sampling_rate of new device and MeasurementChain
+            # is the same
+            if not self.sampling_rate == device_data.sampling_rate:
+                raise ValueError("ref_signal has wrong samping_rate")
+            # add device to chain
+            new_device = DeviceObj(device_data,
+                                   sens,
+                                   device_name)
+            self.devices.append(new_device)
+
+    def list_devices(self):
+        """Returns a list of names of all devices in the measurement chain.
+        """
+        # list all ref-objects in chain
+        device_names = []
+        for dev in self.devices:
+            name = dev.device_name
+            device_names.append(name)
+        return device_names
+
+    def remove_device(self, num):
+        """Removes a single device from the measurement chain,
+         by name or number.
+
+        Attributes
+        ----------
+        num : int or str
+        Identifier for device to remove. Device can be found by name as string
+        or by number in device list as int.
+        """
+        # remove ref-object in chain position num
+        if isinstance(num, int):
+            front = self.devices[:num]
+            back = self.devices[num+1:]
+            for i in back:
+                front.append(i)
+            self.devices = front
+        # remove ref-object in chain by name
+        elif isinstance(num, str):
+            i = 0
+            for dev in self.devices:
+                if dev.name == num:
+                    self.remove_device(i)
+                    return
+                i = i + 1
+            raise ValueError(f"device {num} not found")
+        else:
+            raise TypeError("device to remove must be int or str")
+
+    # reset complete ref-object-list
+    def reset_devices(self):
+        """Resets the list of devices in the measurement chain.
+        Other global parameters such as sampling rate or sound device of the
+        measurement chain remain unchanged.
+        """
+        self.devices = []
+
+    # get the freq-response of specific device in measurement chain
+    def device_freq(self, num):
+        """Returns the frequency response of a single device from the
+        measurement chain, by name or number.
+
+        Attributes
+        ----------
+        num : int or str
+        Identifier for device, can be name as string or by number
+        in device list as int.
+        """
+        if isinstance(num, int):
+            dev = self.devices[num].data * self.devices[num].sens
```

Comment:
This is duplicate code - you should use the method from the device class to get the frequency response